retrystrategies/retry_test.go

```diff
@@ -0,0 +1,39 @@
+package retrystrategies
+
+import (
+	"testing"
+
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRetry_CreatesExponential(t *testing.T) {
+	m := datastore.Metadata{
+		Strategy:        "exponential-backoff",
+		RetryLimit:      20,
+		IntervalSeconds: 5,
+	}
+	var r RetryStrategy = NewRetryStrategyFromMetadata(m)
+	assert.NotEqual(t, r.NextDuration(1), r.NextDuration(3))
```

Comment:
I'm sure there's a way to confirm which implementation gets returned here. But for the life of me, I cant remember how that's done.

I remember it looking like:
`r.(ConcreteImplementation)`

Reply:
resolved in https://github.com/frain-dev/convoy/pull/355/commits/5c2514044a001a8a20a53712ab0264da42df7bbe